Assn3_Pseudo.txt

```diff
@@ -0,0 +1,40 @@
+int main()
+   var plaintext
+   get(plaintext)                                        // get text to be hashed
+   var[] blocks = getBlocks(plaintext.AlNumOnly)         // separate into blocks
+   var[] intBlocks = operation(blocks, AlNumOnly.length) // ?
+   print("Hash: " + fixLength(intBlocks)                 // print hashed value
+
+// create blocks from a shifted bitwise OR of each 4-character chunk
+var[] getBlocks(var plaintext)
+   var chunk = 0
+   var blocks;
+   for (var i = 0; i < plaintext.length; i += 4) // for each chunk
+      chunk = 0
+      for (j = 0 to 4)                           // for each character in chunk
+         chunk = chunk << 8                      // shift chunk left 8 bits
+         if (j + i < plaintext.length)
+            var temp = plaintext(j+i)
+            chunk = chunk | temp                 // bitwise OR with current char
+      blocks.push(chunk)
+   return blocks
+
+// bitwise OR a left and right shift of each block to the power of the total length

```

Comment:
for each block raise it to the power of the plaintext's message length, then perform modulus 32 on resulting value to obtain value for rotational shift.